### ISSUE TITLE
Doesn't compile with gcc 9 

### ISSUE THREAD

**Issue by robinmoussu** [hanickadot/compile-time-regular-expressions#44]
Hello,

I don't know if it's a gcc9-related bug, or a bug in ctre (current master, `91fce35`):

```sh
➜  make
In file included from include/ctre.hpp:4,
                 from tests/range.cpp:1:
include/ctre/literals.hpp:54:1: error: literal operator template ‘constexpr auto ctre::literals::operator""_ctre()’ has invalid parameter list;  Expected non-type template parameter pack <char...>   or single non-type parameter of class type
   54 | }
      | ^
include/ctre/literals.hpp:81:1: error: literal operator template ‘constexpr auto ctre::test_literals::operator""_ctre_test()’ has invalid parameter list;  Expected non-type template parameter pack <char...>   or single non-type parameter of class type
   81 | }
      | ^
include/ctre/literals.hpp:93:1: error: literal operator template ‘constexpr auto ctre::test_literals::operator""_ctre_gen()’ has invalid parameter list;  Expected non-type template parameter pack <char...>   or single non-type parameter of class type
   93 | }
      | ^
include/ctre/literals.hpp:104:1: error: literal operator template ‘constexpr auto ctre::test_literals::operator""_ctre_syntax()’ has invalid parameter list;  Expected non-type template parameter pack <char...>   or single non-type parameter of class type
  104 | }
      | ^
cc1plus: warning: unrecognized command line option ‘-Wno-gnu-string-literal-operator-template’
make: *** [Makefile:31: tests/range.o] Error 1
```

My g++ is really recent (3 days old, from trunk).

```sh
➜  g++ --version
g++ (GCC) 9.0.1 20190126 (experimental)
Copyright © 2019 Free Software Foundation, Inc.
```
---

The last commit that I can compile is `7eff180` (with a lot of warnings).

---

On previous commits (like `e121f1e`), but more recent than `7eff180` I had this error:
```sh
tests/matching2.cpp:192:42: error: unable to find string literal operator ‘operator""sv’ with ‘const char8_t [4]’, ‘long unsigned int’ arguments
  192 | static_assert(CTRE_CREATE("^abc$").match(u8"abc"sv));
```

**Comment by hanickadot** [comment]
Can you send me how you invocate compilation with your GCC?

**Comment by hanickadot** [comment]
If you are using GCC9 you should use `-std=c++2a` mode. Current GCC9 doesn't support template UDLs as specified in https://wg21.link/p0732

You can use c++20 syntax `ctre::match<"pattern">(subject)` or in c++17 `ctre::match<pattern_name>(subject)`, but if you use GNU extension UDLs in gcc9 with C++17 mode you will get warnings in -pedantic. I enabled them in C++17 mode in GCC9 in latest commit.

**Comment by robinmoussu** [comment]
I cloned this repo then did `make` on different commits. I didn't checked what were the flags that are sends to g++. I will check this evening when I will be back home.

**Comment by hanickadot** [comment]
ok, I found the problem (the new head version of GCC is 9.0.1, and I was explicitly checking for 9.0.0), fixed in #2bfa8134228fc46603d4a8b65fd49c0527e44c57

**Comment by robinmoussu** [comment]
Perfect, it's fixed ;)
